openhgnn/sampler/MAGNN_sampler.py

```diff
@@ -0,0 +1,195 @@
+import dgl
+import numpy as np
+from openhgnn.models.MAGNN import mp_instance_sampler, mini_mp_instance_sampler
+
+class MAGNN_sampler():
+    '''
+    Description
+    -----------
+    MAGNN sampler based on the implementation of author. The author only implements one sampling 1-layer subgraphs. Here
+    we implement the MAGNN sampler which can sample n-layer subgraphs based on DGL and Pytorch.
+
+    The MAGNN sampler samples n-layer subgraphs based on seed_node of a certain class, metapath instances of seed_nodes
+    which contain all neighbors that might be sampled.
+
+    Users can make use of the MAGNN sampler as a dataset of torch.DataLoader so that we can assign num_workers to
+    concurrently speed up sampling subgraphs.
+
+    Parameters
+    ----------
+    g : dgl.DGLGraph
+        the dgl graph from which we sample subgraphs.
+    mask : list or numpy.ndarray
+        the train/test/test nodes mask, which should be like train mask = [0, 1, 0, 0, 1] indicating that the
+        1th, 4th nodes are training nodes of g. Note that the mask indicates which part of nodes in g will be sampled
+        by MAGNN sampler. e.g. if it's the test mask, the sample will be employed on testing nodes of g to sample
+        subgraphs for testing phase. This param is useful when the graph is too large to be directly validate on.
```

Comment:
Can it be a list of node IDs?  A list of node IDs should be more efficient than a boolean array if the number of nodes is considerably smaller.

Also what are the node types?

Reply:
The mask here is the mask of the whole graph instead of seed nodes. e.g the training mask of nodes in the graph. Sometimes the mask indicates the phase that sampler is employed on like training or validation phase. The mask here is used to specify which part of nodes in the whole graph to be seed nodes.

Reply:
Yeah you are right.  The modifier "seed" is unnecessary.